src/peer_discovery.cpp

```diff
@@ -154,12 +142,12 @@ PeerDiscovery::m_sendto(uint8_t *buf,size_t &buf_size){
 
 }
 
-void
-PeerDiscovery::m_recvfrom(uint8_t *buf,size_t &buf_size)
+SockAddr
+PeerDiscovery::m_recvfrom(uint8_t *buf,const size_t &buf_size)
```

Comment:
This function or the caller should have a way to check the received byte number, to avoid reading uninitialized memory.